src/main/java/com/spotify/github/v3/search/requests/SearchParameters.java

```diff
@@ -47,4 +50,15 @@ public interface SearchParameters extends Parameters {
 
   /** The sort order if sort parameter is provided. One of asc or desc. Default: desc */
   Optional<String> order();
+
+  /**
+   * The number of results per page (max 100). Default: 30
+   */
+  @SuppressWarnings("checkstyle:methodname")
+  Optional<Integer> per_page();
```

Comment:
I think it makes sense to use the java method naming convention of camelCase names here and instead use the ` @JsonProperty("per_page")` annotation to make it compatible with github's format. WDYT?

Reply:
> I think it makes sense to use the java method naming convention of camelCase names here and instead use the ` @JsonProperty("per_page")` annotation to make it compatible with github's format. WDYT?

I'd like to use camelCase as well, but it turned out that serialization will be broken in this case: https://github.com/spotify/github-java-client/blob/4a61f43c30b90590612e32de5ef54219280ae89a/src/main/java/com/spotify/github/Parameters.java#L57

Here is the method from Parameters.java:

```
/**
   * Goes through all public methods defined in an interface that extends this interface and calls
   * them in the context of the class that called this method, then joins the method name with the
   * result it produced using an ampersand (&amp;) as a delimiter.
   *
   * <p>It works on interfaces with deep inheritance and filters out any methods defined on this
   * interface (with the assumption that they come from the same class loader).
   *
   * @return String of "key=value" joined on &amp;
   */
  default String serialize() {
    return Arrays.stream(this.getClass().getInterfaces())
       ...
        .filter(method -> !method.getDeclaringClass().equals(Parameters.class))
        .collect(
            toMap(
                Method::getName,
                method -> {
                  ...
        .sorted((entry1, entry2) -> entry1.getKey().compareTo(entry2.getKey()))
        .map(entry -> entry.getKey() + "=" + entry.getValue().get())
        .collect(joining("&"));
  }
```

Reply:
Thanks for clarifying. You're right, since we're not dealing with JSON here at all my suggestion won't work. This is fine as-is then 👍